### PR TITLE
Feat/trip stops

### PR DESCRIPTION
**Pull request by AlexKMarshall**

### CODE REVIEW

File: client/src/components/AutoComplete.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        } = item;
+        const formatted = { type, geometry, properties: { name, label } };
+        return formatted;
+      });
+    });
+}
+
+type ControlledAutocompleteProps = {
+  name: string;
+};
```
Comment (hucki) [inline client/src/components/AutoComplete.tsx:33]: will this go into the APIClient later? Or is it intended to stay here?

Comment (AlexKMarshall) [reply client/src/components/AutoComplete.tsx:33]: It'll definitely get moved. We need a good set of type-conversion functions, which I'm investigating now.

File: client/src/types/Trip.ts [hunk 2]
```diff
@@ -4,6 +4,11 @@ export type Trip = {
...
   endDate: Date;
-  stops: string[];
   details: GeoJSON.FeatureCollection;
 };
+
+export type Stop = {
+  type: string;
+  geometry: { type: string; coordinates: number[] };
+  properties: { name: string; label: string };
+};
```
Comment (hucki) [inline client/src/types/Trip.ts:14]: this will help me solve my problem ;-)

Comment (AlexKMarshall) [reply client/src/types/Trip.ts:14]: Don't think this is the final version, but it's a step in the direction

**Pull request merged**